src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -799,20 +799,50 @@ export default class Fms {
      * @param routeString {string}
      * @return {boolean}
      */
-    replaceDepartureProcedure(routeString) {
+    replaceDepartureProcedure(routeString, airportIcao) {
         const routeStringElements = routeString.split(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
 
-        if (routeStringElements.length !== 3) {
+        if (routeStringElements.length > 3) {
             return [false, 'departure procedure format not understood'];
         }
 
+        if (routeStringElements.length === 1) { // RouteString looks like PROC
+            const procedureId = routeStringElements[0];
+            const sidModel = NavigationLibrary.getProcedure(procedureId);
+
+            if (_isNil(sidModel)) {
+                return [false, 'SID name not understood'];
+            }
+
+            const exitPoint = _find(this._fms.waypoints, (waypointModel) => sidModel.hasExit(waypointModel.name));
+
+            if (exitPoint === '') {
+                return [false, `SID ${procedureId.toUpperCase()} doesn't have an exit along our route`];
+            }
+
+            routeStringElements.push(exitPoint);
+        }
+
+        if (routeStringElements.length === 2) { // RouteString looks like PROC.EXIT
+            const runwayModel = this._fms.departureRunwayModel;
+
+            if (_isNil(runwayModel)) {
+                return [false, 'unsure if we can accept that procedure; we don\'t have a runway assignment'];
+            }
+
+            const entryName = airportIcao.toUpperCase() + runwayModel.name;
+
+            routeStringElements.unshift(entryName);
+        }
+
         const procedureId = routeStringElements[1];
 
         if (!NavigationLibrary.hasProcedure(procedureId)) {
             return [false, `unknown procedure "${procedureId}"`];
         }
 
-        const wasSuccessful = this._routeModel.replaceDepartureProcedure(routeString);
+        const newRouteString = routeStringElements.join(PROCEDURE_OR_AIRWAY_SEGMENT_DIVIDER);
+        const wasSuccessful = this._routeModel.replaceDepartureProcedure(newRouteString);
```

Comment:
Please avoid using `new` in variable names. For instances like these, we like doing `nextRouteString`.

Reply:
done